src/triage/component/catwalk/individual_importance/__init__.py

```diff
@@ -140,15 +139,16 @@ def save(self, importance_records, model_id, as_of_date, method_name):
             as_of_date,
             method_name,
         )
-        for importance_record in importance_records:
-            db_object = IndividualImportance(
-                model_id=int(model_id),
-                entity_id=int(importance_record["entity_id"]),
-                as_of_date=as_of_date,
-                feature=importance_record["feature_name"],
-                feature_value=importance_record["feature_value"],
-                method=method_name,
-                importance_score=float(importance_record["score"]),
-            )
-            db_objects.append(db_object)
-        save_db_objects(self.db_engine, db_objects)
+        def generate_records():
+            for importance_record in importance_records:
+                db_object = IndividualImportance(
+                    model_id=int(model_id),
+                    entity_id=int(importance_record["entity_id"]),
+                    as_of_date=as_of_date,
+                    feature=importance_record["feature_name"],
+                    feature_value=importance_record["feature_value"],
+                    method=method_name,
+                    importance_score=float(importance_record["score"]),
+                )
+                yield db_object
+        save_db_objects(self.db_engine, generate_records())
```

Comment:
nothing wrong with this dynamically-defined generator function; though, not everyone loves that sort of thing, and it doesn't seem necessary –

```suggestion
        save_db_objects(self.db_engine, (
            IndividualImportance(
                model_id=int(model_id),
                entity_id=int(importance_record["entity_id"]),
                as_of_date=as_of_date,
                feature=importance_record["feature_name"],
                feature_value=importance_record["feature_value"],
                method=method_name,
                importance_score=float(importance_record["score"]),
            )
            for importance_record in importance_records
        ))
```